Parsers/ASimAuthentication/Parsers/vimAuthenticationPaloAltoCortexDataLake.yaml

```diff
@@ -0,0 +1,140 @@
+      "2", "Low",
+      "3", "Low",
+      "4", "Medium",
+      "5", "Medium",
+      "6", "Medium",
+      "7", "High",
+      "8", "High",
+      "9", "High",
+      "10", "High"
+  ];
+  let parser = (disabled: bool = false, starttime: datetime=datetime(null), endtime: datetime=datetime(null), targetusername_has: string='*') {
+      CommonSecurityLog
+      | where not(disabled)
+          and (isnull(starttime) or TimeGenerated >= starttime)
+          and (isnull(endtime) or TimeGenerated <= endtime)
+          and DeviceVendor == "Palo Alto Networks"
+          and DeviceProduct == "LF"
+          and DeviceEventClassID == "AUTH"
+          and (targetusername_has == "*" or AdditionalExtensions has targetusername_has)
+      | parse-kv AdditionalExtensions as (PanOSSourceDeviceHost:string, PanOSSourceDeviceOSFamily:string, PanOSAuthenticationProtocol: string, PanOSAuthenticatedUserDomain: string, PanOSAuthenticatedUserName:string, PanOSAuthenticatedUserUUID:string, start:string) with (pair_delimiter=";", kv_delimiter="=")
+      | where targetusername_has == "*" or PanOSAuthenticatedUserName has targetusername_has
+      | invoke _ASIM_ResolveDvcFQDN('DeviceName')
+      | invoke _ASIM_ResolveSrcFQDN('PanOSSourceDeviceHost')
+      | lookup EventSeverityLookup on LogSeverity
+      | extend
+          EventStartTime = todatetime(start),
+          SrcIpAddr = coalesce(SourceIP,DeviceCustomIPv6Address2),
+          TargetIpAddr = coalesce(DestinationIP,DeviceCustomIPv6Address3),
+          EventMessage = Message
+      | project-rename
+          EventProduct = DeviceProduct,
+          EventVendor = DeviceVendor,
+          DvcIpAddr = Computer,
+          EventUid = _ItemId,
+          DvcId = DeviceExternalID,
+          EventOriginalResultDetails = Message,
+          EventOriginalSeverity = LogSeverity,
+          EventOriginalType = DeviceEventClassID,
+          EventOriginalUid = ExtID,
+          EventProductVersion = DeviceVersion,
+          LogonProtocol = PanOSAuthenticationProtocol,
+          SrcDvcOs = PanOSSourceDeviceOSFamily,
+          TargetUsername = PanOSAuthenticatedUserName,
+          TargetUserId = PanOSAuthenticatedUserUUID,
+          TargetDomain = PanOSAuthenticatedUserDomain
+      | extend
+          Dvc = coalesce(DvcId, DvcHostname, DvcIpAddr),
+          EventEndTime = EventStartTime,
+          EventResult = iff(EventMessage has "Invalid Certificate", "Failure", "Success"),
+          Dst = TargetIpAddr,
+          Src = coalesce(SrcHostname, SrcIpAddr),
+          TargetUserType = _ASIM_GetUserType(TargetUsername, ""),
+          User = TargetUsername,
+          IpAddr = SrcIpAddr,
+          DvcIdType = iff(isnotempty(DvcId), "Other", ""),
+          TargetDomainType = case(
+                            array_length(split(DestinationUserName,".")) > 1, "FQDN",
+                            array_length(split(DestinationUserName,"\\")) > 1, "Windows",
+                            ""
```

Comment:
we should evaluate TargetDomainType using TargetDomain and not using DestinationUserName

Reply:
@niralishah-crest please set TargetDomainType only when TargetDomain is present is present. TargetDomain is related to device not the user